apps/cowswap-frontend/src/common/containers/ConnectionStatus/index.tsx

```diff
@@ -0,0 +1,83 @@
+import { useSetAtom } from 'jotai'
+import { useCallback, useEffect, useRef } from 'react'
+
+import { useIsOnline } from '@cowprotocol/common-hooks'
+import { useAddSnackbar, removeSnackbarAtom, SnackbarItem } from '@cowprotocol/snackbars'
+import { UI } from '@cowprotocol/ui'
+
+import { AlertTriangle } from 'react-feather'
+import styled from 'styled-components/macro'
+
+const OfflineIcon = styled(AlertTriangle)`
+  color: var(${UI.COLOR_DANGER});
+`
+
+const OFFLINE_NOTIFICATION_ID = 'offline-connection'
+const NOTIFICATION_REFRESH_INTERVAL = 60000 // 1 minute
+const NOTIFICATION_DURATION = NOTIFICATION_REFRESH_INTERVAL + 10000 // Slightly longer than refresh interval
+
+/**
+ * Creates a standardized offline notification object
+ */
+const createOfflineNotification = (): SnackbarItem => ({
+  id: OFFLINE_NOTIFICATION_ID,
+  icon: 'custom',
+  customIcon: <OfflineIcon size={24} />,
+  content: (
+    <div>
+      <strong>Connection Lost</strong>
+      <br />
+      You appear to be offline. Some features may not work properly.
+    </div>
+  ),
+  duration: NOTIFICATION_DURATION,
+})
+
+/**
+ * Custom hook to manage offline notifications with proper cleanup and no race conditions
+ */
+function useOfflineNotification(): void {
+  const isOnline = useIsOnline()
+  const addSnackbar = useAddSnackbar()
+  const removeSnackbar = useSetAtom(removeSnackbarAtom)
```

Comment:
Please, add a hook for that. It's better to not expose atoms from libs (if possible)

Reply:
Addressed.